marquez_airflow/dag.py

```diff
@@ -116,20 +116,24 @@ def handle_callback(self, *args, **kwargs):
             dagrun = args[0]
             task_instances = dagrun.get_task_instances()
             for ti in task_instances:
-                extractor = self._extractors.get(ti.operator.__class__)
-                steps_meta = extractor(ti.operator).extract_on_complete(ti)
-                self.log.info(f'steps_meta: {steps_meta}')
-
-                # FIXME: replace with update api
-                for step in steps_meta:
-                    self.get_marquez_client().create_job(
-                        job_name=step.name, job_type=JobType.BATCH,
-                        location=(step.location or self._get_location(ti)),
-                        input_dataset=step.inputs,
-                        output_dataset=step.outputs,
-                        context=step.context,
-                        description=self.description,
-                        namespace_name=self.marquez_namespace)
+                # TODO: Only apply extract_on_complete() for BigQuery.
+                #  We'll need to revisit the base extractor interface to
+                #  see how it might apply to other operators.
+                if isinstance(ti.operator, BigQueryOperator):
```

Comment:
We should just add a no op extract_on_complete to the BaseOperator instead. This was the intent when this was added to the BigQueryExtractor. It is an optional operation for extractors to implement.

Reply:
Agreed, I was thinking of opening a PR to address the base extractor contract, but makes sense to just do it now 👍